src/bokeh/models/widgets/tables.py

```diff
@@ -858,6 +858,37 @@ def __init__(self, *args, **kwargs) -> None:
     The height of each row in pixels.
     """)
 
+    @staticmethod
+    def to_data_table(data, columns=None, formatters={}, **kwargs) -> DataTable:
+        """ Create a simple table from a pandas dataframe, dictionary or ColumnDataSource.
+
+        Parameters:
+        data (pandas DataFrame or dict or ColumnDataSource): The data to create the table from.
+        columns (list, optional): A list of column names to use. If None, use all columns.
+        formatters (dict, optional): A formatter to be applied to all columns or specified columns.
```

Comment:
AFAIK these must be indented to render properly, also we use `Args` consistently everywhere else I think
```suggestion
        Args:
            data (DataFrame or dict or ColumnDataSource) : The data to create the table from.
            columns (list, optional) : A list of column names to use. If None, use all columns.
            formatters (dict, optional) : Formatters to be applied to specified columns.
```
also changed formatters descriptions, since it seems to only accept only a `dict`

Reply:
Also needs a note like "Additional keyword arguments will be passed to DataTable"

Reply:
We also need to be explicit somewhere about which cases will create a whole new CDS, and which ones won't.